### PR TITLE
include interactive guidance in our contributing guide.

### LINKED ISSUE #4571: Poll: Should runnable samples include output as comments?
[this comment](https://github.com/dotnet/docs/pull/4519#discussion_r172756393) highlights a question we've been wrestling with: should runnable samples include the output as comments or not?

The advantage of including comments is that you can see what code should do without running it. The advantage of not including the comments is that the samples take less space, and after running them, you can see the output and the code without scrolling.

Anytime we've asked, the responses have been almost evenly split. Try these two pages and vote:

[Example where comments are included](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/foreach-in#example) Vote  👎 

[Example without comments for output](https://docs.microsoft.com/en-us/dotnet/api/system.datetime?view=netframework-4.7.1#invoke-constructors) Vote 👍

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #4571

We asked users how they want to work with interactive samples. Whether they want comments, or output included. We learned a lot, and updated our guidance based on the feedback.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
 
 For more information, see the [Samples Readme](https://github.com/dotnet/docs/blob/master/samples/README.md).
 
+## The C# interactive experience #
+
+Short code samples in C# can use the `csharp-interactive` language tag to
+specify a C# sample that run in the browswer. When these sample are built, the code sample
```
Comment (ChrisMaddock) [inline CONTRIBUTING.md]: 'browswer' 😄

Comment (ChrisMaddock) [reply CONTRIBUTING.md]: Also, 'When these sample**s** are built'?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
...
+displays as a code window and an output window. The output window displays any output from
+executing the interactive code once the user has run the sample. 
+
+The C# interactive experience changes how we work with samples. Visitors can run the sample
+to see the results by running the sample. A number of factors help determine if the sample
```
Comment (ChrisMaddock) [inline CONTRIBUTING.md]: "Visitors can run the sample to see the results ~by running the sample~."?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
...
+or corresponding text should include information about the output.
+
+### When to display the expected output without running the sample
+
+- Articles intended for beginners should provide output so that readers can compare the output of their work with the expected answer.
+- Samples where the output is integral to the topic should display that output. For example, articles on text formatting should show  the text format without running the sample.
```
Comment (ChrisMaddock) [inline CONTRIBUTING.md]: nitpick - double space!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
...
+- When both the sample and the expected output is short, consider showing the output. It saves a bit of time.
+- Articles explaining how current culture or invariant culture affect output should explain the expected output. The interactive REPL runs on a Linux based host. Culture-specific output is different for different OSes and machines. The article should explain the output in Windows, Linux, and Mac systems.
```
Comment (ChrisMaddock) [inline CONTRIBUTING.md]: Should this point mention that the culture is taken from the users browser? That's what I understood from your previous post on the issue, but I'm not sure it's so clear here?

File: styleguide/template.md [hunk 5]
```diff
@@ -269,6 +269,7 @@ Use three backticks (\`\`\`) + a language ID to apply language-specific color co
 |AzCopy|azcopy|
 |C++|cpp|
 |C#|csharp|
+|C# in browser|csharp-interactive}
```
Comment (ChrisMaddock) [inline styleguide/template.md]: Accidental curly brace, I think!

**Review by ChrisMaddock — COMMENTED** [review]
Can't wait for this to become more widespread. 😄

File: CONTRIBUTING.md (hunk 4 above)
Comment (ChrisMaddock) [reply CONTRIBUTING.md]: (It may be that I'm wrong here, instead!)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
 
 For more information, see the [Samples Readme](https://github.com/dotnet/docs/blob/master/samples/README.md).
 
+## The C# interactive experience #
+
+Short code samples in C# can use the `csharp-interactive` language tag to
```
Comment (pkulikov) [inline CONTRIBUTING.md]: So far we've used `code-csharp-interactive` to insert runnable C# code. Is `csharp-interactive` one more way?

File: CONTRIBUTING.md (hunk 1 above)
Comment (rpetrusha) [inline CONTRIBUTING.md]: run --> runs
When these sample are built, the code sample --> When such a sample is built, it

File: CONTRIBUTING.md (hunk 3 above)
Comment (rpetrusha) [inline CONTRIBUTING.md]: text format --> formatted text

File: CONTRIBUTING.md (hunk 4 above)
Comment (rpetrusha) [reply CONTRIBUTING.md]: @ChrisMaddock, at some point, culture will be taken from the user's browser. But at the moment, the invariant culture is used.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -118,6 +118,29 @@ Your topic will also contain links to the sample. Link directly to the sample's
...
+
+- Articles intended for beginners should provide output so that readers can compare the output of their work with the expected answer.
+- Samples where the output is integral to the topic should display that output. For example, articles on text formatting should show  the text format without running the sample.
+- When both the sample and the expected output is short, consider showing the output. It saves a bit of time.
+- Articles explaining how current culture or invariant culture affect output should explain the expected output. The interactive REPL runs on a Linux based host. Culture-specific output is different for different OSes and machines. The article should explain the output in Windows, Linux, and Mac systems.
+
+### When to exclude expected output from the sample 
+
+- Articles where the sample generates a larger output should not include that in comments. It obscures the code once the sample has been run.
+- Articles where the sample demonstrates a topic, but the output isn't integral to understanding. For example, code that runs a LINQ query to explain query syntax and then display every item in the output collection.
```
Comment (rpetrusha) [inline CONTRIBUTING.md]: understanding --> understanding it

**Review by rpetrusha — APPROVED** [review]
Generally looks good, @BillWagner. @ChrisMaddock has suggested some changes and corrections, as have I. Once you address them, you can mark this as ready to merge.

File: styleguide/template.md [hunk 8]
```diff
@@ -285,6 +286,8 @@ Use three backticks (\`\`\`) + a language ID to apply language-specific color co
 |XAML|xaml|
 |XML|xml|
 
+The `csharp-interactive` name specifies the C# language, and the ability to run the samples from the browser. These snippets are compiled and executed in a Docker container, and the results of that program execution are displayed in the user's browser window.
```
Comment (pkulikov) [inline styleguide/template.md]: Should this paragraph be at the [section](https://github.com/dotnet/docs/blob/master/styleguide/template.md#code) above and mention `code-csharp-interactive`?

File: CONTRIBUTING.md (hunk 4 above)
Comment (ChrisMaddock) [reply CONTRIBUTING.md]: Ah ok. Thanks!

Comment (pkulikov) [inline CONTRIBUTING.md]: I  read *Culture-specific output is different for different OSes and machines* as formatting with the same culture (even invariant) produces different results on different OSes and machines. How much is it true? Does setting current culture to a certain value within the code snippet make output more independent from the host?

**Review by mairaw — COMMENTED** [review]
Folks already got most of the comments I'd make covered. Added just one more.

File: styleguide/template.md (hunk 8 above)
Comment (BillWagner) [reply styleguide/template.md]: I think it fits better here, after the table with all the language tags.

**Pull request merged**